tests/qemu/linux/test_qemu_platform_detail.cpp

```diff
@@ -0,0 +1,202 @@
+
+    mpt::MockLogger::Scope logger_scope = mpt::MockLogger::inject();
+};
+} // namespace
+
+TEST_F(QemuPlatformDetail, ctor_sets_up_expected_virtual_switch)
+{
+    const QString qstring_subnet{QString::fromStdString(subnet)};
+
+    EXPECT_CALL(*mock_utils, run_cmd_for_status(QString("ip"),
+                                                ElementsAre(QString("link"), QString("add"), multipass_bridge_name,
+                                                            QString("address"), _, QString("type"), QString("bridge")),
+                                                _))
+        .WillOnce(Return(true));
+    EXPECT_CALL(*mock_utils, run_cmd_for_status(QString("ip"),
+                                                ElementsAre(QString("address"), QString("add"),
+                                                            QString("%1.1/24").arg(qstring_subnet), QString("dev"),
+                                                            multipass_bridge_name, "broadcast",
+                                                            QString("%1.255").arg(qstring_subnet)),
+                                                _))
+        .WillOnce(Return(true));
+    EXPECT_CALL(*mock_utils, run_cmd_for_status(
+                                 QString("ip"),
+                                 ElementsAre(QString("link"), QString("set"), multipass_bridge_name, QString("up")), _))
+        .WillOnce(Return(true));
+
+    mp::QemuPlatformDetail qemu_platform_detail{data_dir.path()};
+}
+
+TEST_F(QemuPlatformDetail, get_ip_for_returns_expected_info)
+{
+    const mp::IPAddress ip_address{fmt::format("{}.5", subnet)};
+
+    EXPECT_CALL(*mock_dnsmasq_server, get_ip_for(hw_addr)).WillOnce([&ip_address](auto...) { return ip_address; });
+
+    mp::QemuPlatformDetail qemu_platform_detail{data_dir.path()};
+
+    auto addr = qemu_platform_detail.get_ip_for(hw_addr);
+
+    EXPECT_EQ(*addr, ip_address);
+}
+
+TEST_F(QemuPlatformDetail, platform_args_generate_net_resources_removes_works_as_expected)
+{
+    mp::VirtualMachineDescription vm_desc;
+    vm_desc.vm_name = "foo";
+    vm_desc.default_mac_address = hw_addr;
+
+    QString tap_name;
+
+    EXPECT_CALL(*mock_dnsmasq_server, release_mac(hw_addr)).WillOnce(Return());
+
+    EXPECT_CALL(
+        *mock_utils,
+        run_cmd_for_status(QString("ip"),
+                           ElementsAre(QString("addr"), QString("show"), mpt::match_qstring(StartsWith("tap-"))), _))
+        .WillOnce([&tap_name](auto& cmd, auto& opts, auto...) {
+            tap_name = opts.last();
+            return false;
+        });
+
+    mp::QemuPlatformDetail qemu_platform_detail{data_dir.path()};
+
+    const auto qemu_args = qemu_platform_detail.qemu_platform_args(vm_desc);
+
+    EXPECT_TRUE(qemu_args.contains("--enable-kvm"));
+    EXPECT_TRUE(qemu_args.contains("-cpu"));
+    EXPECT_TRUE(qemu_args.contains("host"));
+    EXPECT_TRUE(qemu_args.contains("-nic"));
```

Comment:
Any way we could ensure that it's `--enable-kvm -cpu host` and not `--enable-kvm host -cpu`?